src/Http/Routing/src/Patterns/RoutePatternFactory.cs

```diff
@@ -1186,6 +1152,42 @@ static Action<RoutePatternParameterPart> CheckDuplicateParameters(int parameterC
         return new RoutePattern(rawText, defaults, parameterPolicies, requiredValues, parameters, pathSegments);
     }
 
+    internal static IReadOnlyList<T> CombineLists<T>(
+        IReadOnlyList<T> leftList,
+        IReadOnlyList<T> rightList,
+        Func<int, string, Action<T>>? checkDuplicates = null,
+        string? rawText = null)
+    {
+        if (leftList.Count is 0)
+        {
+            return rightList;
+        }
+        if (rightList.Count is 0)
+        {
+            return leftList;
+        }
+
+        var leftCount = leftList.Count;
```

Comment:
Is this saving something? If it is we should order it as so:
```
var leftCount = leftList.Count;
if (leftCount is 0)
{
    return rightList;
}
var rightCount = rightList.Count;
if (rightCount is 0)
{
    return leftList;
}
```

Reply:
Not really. I was just referencing `leftList.Count` and `rightList.Count` a lot (3 times each) below so this looked a little cleaner to me. I'll adjust it to save the extra getter call. Technically, it could be slow given a bad implementation, but I doubt it is in practice.